tests/cl-go-client/main.go

```diff
@@ -142,32 +152,84 @@ func main() {
+
+	runMessageWithRetries(func() error {
+		_, _, _, _, err := createPosition(igniteClient, expectedPoolId, accountName, lowerTick, upperTick, tokensDesired, defaultMinAmount, defaultMinAmount)
+		return err
+	})
+
+	return accountName, lowerTick, upperTick, tokensDesired, nil
+}
+
+func createManyRandomPositions(igniteClient cosmosclient.Client, poolId uint64, numPositions int) error {
 	for i := 0; i < numPositions; i++ {
-		var (
-			// 1 to 9. These are localosmosis keyring test accounts with names such as:
-			// lo-test1
-			// lo-test2
-			// ...
-			randAccountNum = rand.Intn(8) + 1
-			accountName    = fmt.Sprintf("%s%d", accountNamePrefix, randAccountNum)
-			// minTick <= lowerTick <= upperTick
-			lowerTick = roundTickDown(rand.Int63n(maxTick-minTick+1)+minTick, tickSpacing)
-			// lowerTick <= upperTick <= maxTick
-			upperTick = roundTickDown(maxTick-rand.Int63n(int64(math.Abs(float64(maxTick-lowerTick)))), tickSpacing)
-
-			tokenDesired0 = sdk.NewCoin(denom0, sdk.NewInt(rand.Int63n(maxAmountDeposited)))
-			tokenDesired1 = sdk.NewCoin(denom1, sdk.NewInt(rand.Int63n(maxAmountDeposited)))
-			tokensDesired = sdk.NewCoins(tokenDesired0, tokenDesired1)
-		)
+		_, _, _, _, err := createRandomPosition(igniteClient, poolId)
+		if err != nil {
+			// Handle the error
+			return err
+		}
+	}
+	return nil
+}
+
+func createManyRandomPositionsAndAddToIt(igniteClient cosmosclient.Client, poolId uint64, numPositions int) error {
+	for i := 0; i < numPositions; i++ {
+		accountName, lowerTick, upperTick, tokensDesired, err := createRandomPosition(igniteClient, poolId)
+		if err != nil {
+			return err
+		}
+
+		// Add to position
+		randAmt0 := sdk.NewInt(rand.Int63n(maxAmountDeposited))
+		randAmt1 := sdk.NewInt(rand.Int63n(maxAmountDeposited))
+
+		runMessageWithRetries(func() error {
+			positionId, _, _, _, err := createPosition(igniteClient, expectedPoolId, accountName, lowerTick, upperTick, tokensDesired, defaultMinAmount, defaultMinAmount)
+			if err != nil {
+				return err
+			}
+
+			_, _, _, err = addToPosition(igniteClient, positionId, accountName, randAmt0, randAmt1, defaultMinAmount, defaultMinAmount)
+			return err
+		})
+	}
+
+	return nil
+}
+
+func createManyRandomPositionsToWithdraw(igniteClient cosmosclient.Client, poolId uint64, numPositions int) error {
+	for i := 0; i < numPositions; i++ {
+		accountName, lowerTick, upperTick, tokensDesired, err := createRandomPosition(igniteClient, poolId)
+		if err != nil {
+			return err
+		}
 
 		runMessageWithRetries(func() error {
-			_, _, _, err := createPosition(igniteClient, expectedPoolId, accountName, lowerTick, upperTick, tokensDesired, defaultMinAmount, defaultMinAmount)
+			positionId, _, _, _, err := createPosition(igniteClient, expectedPoolId, accountName, lowerTick, upperTick, tokensDesired, defaultMinAmount, defaultMinAmount)
+			if err != nil {
+				return err
+			}
+
+			randAmt := sdk.NewInt(rand.Int63n(maxAmountDeposited))
```

Comment:
note: i originally thought of using `liquidityDelta` but got lots of overflows(more liquidity to withdraw than expected) so i used `maxAmountDeposited` which is the original amount that we use to createPosition